### ISSUE TITLE
No Audio V1.7.0

### ISSUE THREAD

**Issue by Daddio61** [moneriomaa/openear#57]
Hi, thank you for a great program.
It seems that I've stumbled across a repeatable fault in the audio setup/path area.

Windows 10 Pro build 19041.508 (up to date)
OpenEar v1.6.0 works with audio
OpenEar v1.7.0 DOES NOT WORK WITH AUDIO. (Repeatable fault)
OpenEar v1.7.0 has way too many options in the dropdown box (3 entries each of Realtek Speakers, VB Cable Input, VB Cable output, VAC Line 1, Line 2, and Line 3 outputs and inputs !!!!

Computer output is Realtek Speakers, also set in OpenEar v1.7.0.
Program and computer are not muted.
Volume in computer and program set to comfortable listening level.

No sound played and no levels shown in Sound Control Panel when OpenEar v1.7.0 tuned to a known FM broadcast station. (Works with openEar v1.6.0)
Sound playback tab shows no levels for speakers.

I selected VB Cable Input FOR A LAUGH!!!
Sound playback audible and shows a level with OpenEar v1.7.0 in VB Cable Input, and VB Cable Output (WHEN SELECTED).
WxtoImg APT decoding program is connected to VB Cable Output!!!!

Selected App Volume and Device Preferences in Windows Sound Setting main page.
AFTER RESETTING WINDOWS SOUNDS, OpenEar v 1.7.0 commenced working EVEN THOUGH OUTPUT still VB Cable Input!!
Changed OpenEar sound output to Realtek Speakers and sounds continued to play!
Closed OpenEar v 1.7.0 down and restarted the program.
No audio output again!!
Reset Audio again in Windows Sound Settings as above and works again!!

I daily receive and decode NOAA APT weather satellites whilst playing on the computer and no sound paths are linked. I use  SDR# v1740, VB Cable Input, VB Cable Output into WxtoImg to achieve this.

I believe the error might be coming initially from the multiple entries in the OpenEar v1.7.0 dropdown SOUND box 

I hope you can help me with this.  If there are any files that I can send regarding this just ask.

Graham

**Comment by JoeyAbsi** [comment]
I have the problem as well:

V1.6.0 is working normally
V1.7.0 there is no audio although signal decoding seems to be working.

I don't know how to fix this issue though.

**Comment by moneriomaa** [comment]
in version 1.7.x there is option as audio output added to software
that makes for some users some troubles for listening to audio output
i think it is about incorrect settings of audio device for output, it depends to both windows sound settings and option was selected on OpenEar

one way for checking the settings and output selection is correct:
opening sound settings on windows and selecting tab playback devices
then user must check when software runs and demodulates or decodes which of devices green level raised
if device is what the user expects (hearable device like speakers), then user must select this on output list in openear

**Comment by Daddio61** [comment]
Yes I tried that, and the level indicator was against the VB Cable input!!!! And with NO SPEAKERS connected to the output, nor the LISTEN TO THIS DEVICE checked, OpenEar played sound out of my speakers!!!!

Not what I require!! No other program ruins my Windows Sound settings like OpenEar v1.70.

I use VB Cable to pipe audio from another RTL SDR into a decoding program that runs 24/7/365. 

Having OpenEar v1.70 push its audio into that stream somehow is useless to me. 

Graham Hunter
Cooloola Cove 4580

grahamhunter61@gmail.com

**Comment by JoeyAbsi** [comment]
Yeah I tried selecting different output devices and it eventually works (not on specific output). 
Sometimes it works on Speaker 1, other times it's on speaker 2 or VB Cable input etc.

The good thing it works but it is still a bug in the program.

**Comment by hayden-t** [comment]
same here https://github.com/moneriomaa/openear/issues/56

**Comment by moneriomaa** [comment]
i think it is about sound settings
in version 1.6.0 output set to default output device then user can listen to sound
but in version 1.7.0 user must select output device
and for this there is two parameter one for windows sound device selected and two OpenEar output selected
you can open windows sound setting from control panel and run openear and play FM for example see which level be green on sound settings playback devices
it must hearable device like speakers if other device be green then change output selection on openear to hearable device be green level
finally selected device on output and green level on sound settings must be same name device

**Comment by Daddio61** [comment]
I’ve done that, however OpenEar v1.7 says I have 3speakers, 3 VB Cable Inputs, 3 VAC 1 Inputs etc!!!  No other program using audio displays three times the number of audio inputs and outputs as my computer really has!!!

No other program using audio connects my audio INPUTS  as, or to my OUTPUTS.

NO other program STOPS me using VB Cable INPUT to VB Cable OUTPUT . I can’t use OpenEar AND decode weather satellites otherwise WxtoImg gets OpenEar audio AND the audio from my other RTL SDR!!!

Graham Hunter
Cooloola Cove 4580

grahamhunter61@gmail.com

**Comment by moneriomaa** [comment]
**_I’ve done that, however OpenEar v1.7 says I have 3speakers, 3 VB Cable Inputs, 3 VAC 1 Inputs etc!!!  No other program using audio displays three times the number of audio inputs and outputs as my computer really has!!!_**
Because no program does that, so shouldn't another program?
I don't say openear is program without bug or issue, but I  think your argument is not correct
about repeated device issue, it is the windows answer when i read devices from it! and those name is different in list but yes maybe same device listed in different name but as i said it is  windows answer and maybe different and maybe same! i am not sure, on next release i try remove repeated answer from windows if they are same 

**_No other program using audio connects my audio INPUTS  as, or to my OUTPUTS._**
it is not about openear, it is maybe about your sound settings(maybe you checked "listen to this device" on recording device)

**_NO other program STOPS me using VB Cable INPUT to VB Cable OUTPUT . I can’t use OpenEar AND decode weather satellites otherwise WxtoImg gets OpenEar audio AND the audio from my other RTL SDR!!!_**
it is your side state and it is not cleared for me, many reason maybe led to it! every program works with its state and requirement, but maybe need more test by other programs to detect what is wrong, some times something looks wrong but in fact some bad settings or usage led to it, but some times bugs led to it, it will be clear on time and more testing

**Comment by Daddio61** [comment]
I can run FLdigi, JTDX, JAERO, MultiPSK, Scytale-C, DSD+ freeware version, WxtoImg, and all report the correct number of audio inputs and outputs.
I NEVER have “Listen to this device” in recording section checked. 

I’m just trying to supply all the relevant data for you so that you can appreciate what is happening here; it’s not like you can drop in and see for yourself.

I really do value and appreciate the efforts you have made in programming OpenEar, I’m not a programmer but I am a competent computer user. 

Let’s wait for the next version.



Graham Hunter
Cooloola Cove 4580

grahamhunter61@gmail.com

**Comment by moneriomaa** [comment]
yes, i got the users feedbacks goal and really thanks from all
but about version 1.7.0 output sound issue, really i am not sure, because many users use this with correct settings
like me, i really easily use it on my side on 3 test system, but for users that has issue on voice output i try answer them one by one, but some times users send their feedback without read my answer on same issue! then repeatable answer i wrote.
about conflicting openear with other program like weather program, yes i say maybe some issue there but need test with same state of you in my side (i try to run weather program in same time with openear) to detect what is wrong! about repeated device on list yes i know it is not desired but usually i released new version with major changes, i try to remove repeated on next release
